source/dofs/dof_tools_constraints.cc

```diff
@@ -1070,6 +1101,1290 @@ namespace DoFTools
+              // of these lines is shown above (see
+              // n_internal_lines_on_children)
+              for (unsigned int dof = 0; dof < fe.n_dofs_per_line(); ++dof)
+                dofs_on_children.push_back(
+                  this_face->child(0)->line(1)->dof_index(dof, fe_index));
+
+              for (unsigned int dof = 0; dof < fe.n_dofs_per_line(); ++dof)
+                dofs_on_children.push_back(
+                  this_face->child(2)->line(1)->dof_index(dof, fe_index));
+
+              for (unsigned int dof = 0; dof < fe.n_dofs_per_line(); ++dof)
+                dofs_on_children.push_back(
+                  this_face->child(0)->line(3)->dof_index(dof, fe_index));
+
+              for (unsigned int dof = 0; dof < fe.n_dofs_per_line(); ++dof)
+                dofs_on_children.push_back(
+                  this_face->child(1)->line(3)->dof_index(dof, fe_index));
+
+              // DoFs on the bordering lines:
+              // DoFs on the exterior lines to the children; the order of
+              // these lines is shown above (see n_external_lines_on_children)
+              for (unsigned int line = 0;
+                   line < GeometryInfo<dim>::lines_per_face;
+                   ++line)
+                for (unsigned int child = 0; child < n_children_per_line;
+                     ++child)
+                  for (unsigned int dof = 0; dof < fe.n_dofs_per_line(); ++dof)
+                    dofs_on_children.push_back(
+                      this_face->line(line)->child(child)->dof_index(dof,
+                                                                     fe_index));
+
+              // DoFs on the faces of the four children:
+              for (unsigned int child = 0; child < n_children_per_face; ++child)
+                {
+                  // skip artificial cells
+                  if (cell->neighbor_child_on_subface(face, child)
+                        ->is_artificial())
+                    continue;
+
+                  for (unsigned int dof = 0; dof < fe.n_dofs_per_quad(face);
+                       ++dof)
+                    dofs_on_children.push_back(
+                      this_face->child(child)->dof_index(dof, fe_index));
+                } // rof: child
+
+              // consistency check:
+              // note: we can get fewer DoFs when we have artificial cells
+              Assert(dofs_on_children.size() <= n_dofs_on_children,
+                     ExcInternalError());
+
+              // As the Nedelec elements are oriented, we need to take care of
+              // the orientation of the lines.
+              // Remark: "false" indicates the line is not flipped.
+              //         "true" indicates the line is flipped.
+
+              // Orientation - Lines:
+              // get the orientation from the edges from the mother cell
+              std::vector<bool> direction_mother(
+                GeometryInfo<dim>::lines_per_face, false);
```

Comment:
I think you used `int` with value 1 or -1 for the 2D case above for `direction_mother`. Can you comment on the different choices, or unify to a similar data type in case they have the same underlying switches?

Reply:
There is no real reason for that difference. Therefore, I unified it.
The direction is now always indicated by a `bool`.